src/imitation/util/networks.py

```diff
@@ -117,6 +117,57 @@ def forward(self, x: th.Tensor) -> th.Tensor:
         return (x - self.running_mean) / th.sqrt(self.running_var + self.eps)
 
 
+class EMARunningNorm(RunningNorm):
+    """Similar to RunningNorm but uses an exponentially weighting."""
+
+    running_mean: th.Tensor
+    running_var: th.Tensor
+
+    def __init__(
+        self,
+        num_features: int,
+        lamb: float = 0.05,
+        r: float = 0.05,
+        eps: float = 1e-5,
+    ):
+        """Builds EMARunningNorm.
+
+        Args:
+            num_features: Number of features; the length of the non-batch dim.
+            lamb: value between (0, 1) update rate for the mean
+            r: value between (0, 1) update rate for the variance
+            eps: small constant for for numerical stability.
+        """
+        super().__init__(num_features, eps=eps)
+        self.lamb = lamb
+        self.r = r
+
+    def update_stats(self, batch: th.Tensor) -> None:
+        """Update `self.running_mean` and `self.running_var`.
+
+        Uses Macgregor & Harris (1993), "The Exponentially Weighted Moving Variance".
+
+        Args:
+            batch: A batch of data to use to update the running mean and variance.
+        """
+        if len(batch.shape) == 1:  # Assume list is a batch of 1-d vectors
+            batch = batch.unsqueeze(-1)
+
+        b_size = batch.shape[0]
+        b_mean = th.mean(batch, dim=0)
+        self.running_mean *= 1 - self.lamb
+        self.running_mean += b_mean * self.lamb
+
+        b_var = (
```

Comment:
Consider adding some intermediate variable, e.g:

```python
centered = batch - self.running_mean.unsqueeze(0)
```

Won't take up more LOC (as black already forces you to put line 162 on its own line), and I think will make it more readable if step-by-step.